pdns/protobuf.cc

```diff
@@ -97,6 +97,57 @@ void DNSProtoBufMessage::setEDNSSubnet(const Netmask& subnet, uint8_t mask)
 #endif /* HAVE_PROTOBUF */
 }
 
+// ----------------------------------------------------------------------------
+// GCA - Seth Ornstein - 5/30/2017 - extra protobuf information
+// ----------------------------------------------------------------------------
+
+void DNSProtoBufMessage::addTags(const std::string& strLabel, const std::string& strValue)
+{
+#ifdef HAVE_PROTOBUF
+
+  PBDNSMessage_DNSResponse* response = d_message.mutable_response();
+  if (!response)
+    return;
+
+  std::string strTag;
+  strTag  = strLabel;
+  strTag += ",";                           // comma separator between label and value
+  strTag += strValue;
+
+  response->add_tags(strTag);
+
+#endif /* HAVE_PROTOBUF */
+}
+
+void DNSProtoBufMessage::addRRs(const std::string& strName)
+{
+#ifdef HAVE_PROTOBUF
+
+  PBDNSMessage_DNSResponse* response = d_message.mutable_response();
+  if (!response)
+    return;
+
+  PBDNSMessage_DNSResponse_DNSRR* rr = response->add_rrs();
+      if (rr) {
+        string blob;
+        rr->set_name(strName.c_str());
+        rr->set_type(1);
+        rr->set_class_(1);
+        rr->set_ttl(123);
+        char cTemp[4];
+        cTemp[0] = 127;
+        cTemp[1] = 0;
+        cTemp[2] = 0;
+        cTemp[3] = 1;
+        rr->set_rdata(cTemp, 4);
+      }
```

Comment:
This method only adds one RR, it should probably called `addRR()` instead. should Please don't hardcode the type, class, TTL and content here, those should be parameters passed to this method.